src/components/ui/token-input.tsx

```
@@ -23,8 +23,22 @@ function TokenInput({
       return;
     }
 
-    const numericValue = inputValue.replace(/[^0-9]/g, '');
-    const finalValue = numericValue ? Number(numericValue) : null;
+    // Allow numbers and decimal point, limit to 4 decimal places
+    const numericValue = inputValue.replace(/[^0-9.]/g, '');
+
+    // Ensure only one decimal point
+    const parts = numericValue.split('.');
+    const formattedValue = parts.length > 2
```

ESLint

[error] Insert `⏎·····`

(prettier/prettier)